code/game/objects/items/weapons/teleportation.dm

```diff
@@ -57,11 +57,17 @@ Frequency:
 			var/turf/sr = get_turf(src)
 
 			if (sr)
+				var/list/found_beacons = list()
+				var/list/extra_signals = list()
+				for (var/atom/B in get_valid_teleporter_beacons())
```

Comment:
Typing the loop var does an implicit type check, which in this case is unnecessary as you type check in the main loop; would suggest just doing `var/B`.